### PR TITLE
Plushie Update!

### PR DESCRIPTION
**Pull request by FoxxoTrystan**
# Description

By multiple requests i bring my PR from Frontier The [PLUSHIE UPDATE!](https://github.com/new-frontiers-14/frontier-station-14/pull/886)

This PR add a prize vendor machine where you can trade prize ticket for a toy/plushie, to get ticket you win those in arcade machines.

This PR will give a chance for everyone to get the toys they need in a special machine and will make plushies in general more random and fun, in term of random spawns.

WARNING THIS PR DOES NOT MAP THE PRIZE VENDOR AND WILL MOSTLY NEED TO BE MAPPED.

---

# TODO

- [x] Prize Vendor
- [x] Prize Tickets
- [x] New Toys/Plushies
- [x] Prizes

---

<details><summary><h1>Media</h1></summary>
<p>

image

</p>
</details>

---

# Changelog

:cl:
- add: A ton of new plushies has arrived
- add: Prize counter are here, get your prizes by winning games and getting prize tickets! and find prize balls!!

### CODE REVIEW

File: Resources/Prototypes/Entities/Objects/Fun/prizeticket.yml [hunk 1]
```diff
@@ -0,0 +1,395 @@
...
+  parent: PresentRandom
+  id: PrizeBall
+  name: prize ball
+  description: I wounder whats inside!
+  components:
+  - type: Sprite
+    sprite: Objects/Fun/prizeticket.rsi
+    layers:
+      - state: prizeball
+  - type: SpawnItemsOnUse
```
Comment (DangerRevolution) [inline Resources/Prototypes/Entities/Objects/Fun/prizeticket.yml:105]: This is gross.

**Review by DangerRevolution — CHANGES_REQUESTED** [review]
Plushies should be put into antag objectives

File: Resources/Prototypes/Entities/Objects/Fun/prizeticket.yml (hunk 1 above)
Comment (FoxxoTrystan) [reply Resources/Prototypes/Entities/Objects/Fun/prizeticket.yml:105]: This is how Presents is done sadly i cannot spawn an ent that is just a random plushie as long i would love to have an ent like this... actully could i?

Is it possible to make an ent on init that just get a random plushie?

**Comment by FoxxoTrystan** [comment]
> Plushies should be put into antag objectives

Oh Yee.

**Review by DangerRevolution — APPROVED** [review]

**Review by VMSolidus — DISMISSED** [review]

File: Resources/Migrations/migration.yml
Comment (DangerRevolution) [inline Resources/Migrations/migration.yml:1]: this needs to go into eeMigrations.yml

Debug made us a multi-file migration system and we'll make use of it, plus no one wants to fix migration merge conflicts.

**Review by DangerRevolution — CHANGES_REQUESTED** [review]

File: Resources/Migrations/migration.yml
Comment (FoxxoTrystan) [reply Resources/Migrations/migration.yml:1]: Theres no thing as an eeMigrations, time to create it but also that mean VM code was bad.

**Review by VMSolidus — APPROVED** [review]

File: Resources/Prototypes/Entities/Objects/Fun/puppet.yml [hunk 2]
```diff
@@ -40,3 +40,92 @@
...
+  id: BasePetRock
+  name: pet rock
+  description: Your new and only best friend!
+  abstract: true
+  components:
+  - type: DamageOnLand
+    damage:
+      types:
+        Blunt: 1
+  - type: DamageOtherOnHit
```
Comment (DangerRevolution) [inline Resources/Prototypes/Entities/Objects/Fun/puppet.yml:81]: Lol, go go gadget pet rock!

File: Resources/Prototypes/Entities/Objects/Fun/toys.yml [hunk 3]
```diff
@@ -819,6 +819,368 @@
...
 
+- type: entity
+  parent: BasePlushie
+  id: PlushieJester #Weh!
+  name: jester plushie
+  description: A dubious little creature getting up to mischief.
+  components:
+  - type: Sprite
+    sprite: Objects/Fun/toys.rsi
+    state: plushie_jester
```
Comment (DangerRevolution) [inline Resources/Prototypes/Entities/Objects/Fun/toys.yml:830]: I'm assuming this is the frontier admin, I like him so this is cool

**Review by DangerRevolution — APPROVED** [review]
Haven't playtested it, gonna yolo merge it

**Pull request merged**